jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/http/DslCookieManager.java

```diff
@@ -33,10 +64,22 @@ public DslCookieManager disable() {
     return this;
   }
 
+  public DslCookieManager setClearCookiesBetweenIterations(boolean clear) {
+    this.clearEachIteration = clear;
+    return this;
+  }
+
+  public DslCookieManager setCookiePolicy(DslCookieManager.CookiePolicy policy) {
+
+    this.cookiePolicy = policy.getCookiePolicy();
+    return this;
+  }
+
   @Override
   protected TestElement buildTestElement() {
     CookieManager ret = new CookieManager();
-    ret.setClearEachIteration(true);
+    ret.setClearEachIteration(clearEachIteration);
+    ret.setCookiePolicy(cookiePolicy);
```

Comment:
Consider also adding conversion logic for these 2 properties so, if some jmx test plan uses a non default setting, it is properly converted.

Reply:
is there an example of this somewhere i can look at?

Reply:
Sure, you can check for example [DslCsvDataSet Builder](https://github.com/abstracta/jmeter-java-dsl/blob/31c3391ebf4cd70c74429f43e5c1ae9dbff094d1/jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/configs/DslCsvDataSet.java#L318).

In this case it should be something like this (since you need to inherit existing logic in AutoEnabledHttpConfigElement):

```java
@Override
protected MethodCall buildMethodCall(T testElement, MethodCallContext context) {
  // the string is the prefix JMeter uses for jmeter properties when storing in JMX. Is actually optional but when you have multiple properties with same prefix is handy to set it here and don't have to specify it in each property
  TestElementParamBuilder paramBuilder = new TestElementParamBuilder(testElement, "CookieManager");
  // the string here corresponds to the jmeter property name in JMX (removing the prefix if you already added it in param builder)
  MethodParam clearBetweenIterations = paramBuilder.boolParam("clearEachIteration", true);
  MethodParam policy = paramBuilder.enumParam("policy", CookiePolicy.STANDARD);
  if (!clearBetweenIterations.isDefault() || !policy.isDefault()) {
    return buildMethodCall()
        // this string is the name of the method that allows setting the property
        .chain("cleanBetweenIterations", clearBetweenIterations)
        .chain("cookiePolicy", policy);
  } else {
    return super.buildMethodCall(testElement, context);
  }
}
```

And making `CookiePolicy` enum implement `EnumPropertyValue`;

You can check if this properly works by running the tests you have already implemented. Have you ran those tests? Because, since there is no logic to support conversion, they should fail.

Reply:
The comments are just added for your reference. If you need more examples of codeBuilders you can check the class hierarchy of `MethodCallBuilder` in your IDE and checking how different implementations are made. Some are pretty simple and straight forward, some others are more complex, depending on how direct the transformation from JMX to DSL is :).